index.js

```diff
@@ -0,0 +1,232 @@
+const core = require("@actions/core");
+const exec = require("@actions/exec");
+const tc = require("@actions/tool-cache");
+const os = require("os");
+const path = require("path");
+const fs = require("fs");
+
+async function run() {
+  try {
+    // Get inputs
+    const version = core.getInput("esp_idf_version");
+    let idfPath = core.getInput("esp_idf_path");
+    let toolsPath = core.getInput("esp_idf_tools_path");
+
+    // Set default paths if not provided
+    if (!idfPath) {
+      idfPath = process.platform === "win32" ? "C:\\esp\\idf" : "/tmp/esp/idf";
+    }
+    if (!toolsPath) {
+      toolsPath = process.platform === "win32" ? "C:\\esp" : "/tmp/esp";
+    }
+
+    // Install platform-specific dependencies
+    await installDependencies(process.platform);
+
+    // Get the appropriate EIM download URL
+    const eimVersion = "v0.1.5";
```

Comment:
Should this also come from some sort of an argument? If this version is hardcoded, I wonder what the update process would be.

Reply:
Changed to fetch the information from dl.espressif.com and use latest release